File: C#/CreationalDesignPattern/CreationalDesignPattern/Refactoring/Builder/Builders/Builder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using CreationalDesignPattern.Refactoring.Builder.Cars;
using CreationalDesignPattern.Refactoring.Builder.Components;

namespace CreationalDesignPattern.Refactoring.Builder.Builders
{
    public interface Builder
    {
        void SetCarType(CarType type);

        void SetSeats(int seats);

        void SetEngine(Engine engine);

        void SetTransmission(Tranmission tranmission);

        void SetTripComputer(TripComputer tripComputer);

        void SetGPSNavigator(GPSNavigator gpsNavigator);
    }
}

File: Java/CreationalDesignPattern/src/main/java/com/ute/refactoring/builder/builders/Builder.java
package com.ute.refactoring.builder.builders;

import com.ute.refactoring.builder.cars.CarType;
import com.ute.refactoring.builder.components.Engine;
import com.ute.refactoring.builder.components.GPSNavigator;
import com.ute.refactoring.builder.components.Transmission;
import com.ute.refactoring.builder.components.TripComputer;

public interface Builder {
    void setCarType(CarType type);
    void setSeats(int seats);

    void setEngine(Engine engine);
    void setTransmission(Transmission transmission);
    void setTripComputer(TripComputer tripComputer);
    void setGPSNavigator(GPSNavigator gpsNavigator);
}
